### PR TITLE
feat(aws): add how-to find images on AWS

### PR DESCRIPTION
**Pull request by toabctl**
That is basically a copy of [0] with some improvements & corrections.

Refs:
- [0] https://discourse.ubuntu.com/t/cloud-images-amazon-ec2/20337

### CODE REVIEW

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+AWS Marketplace
+---------------
+AWS Marketplace is a digital catalogue with thousands of software listings
+from independent software vendors that make it easy to find, test, buy, and
+deploy software that runs on AWS.
+Canonical maintains `image listings <https://aws.amazon.com/marketplace/seller-profile?id=565feec9-3d43-413e-9760-c651546613f2>`_ for recent Ubuntu releases and special flavors (e.g. `Anbox`, `Pro`, `EKS`) on AWS Marketplace.
+
+Customers can also use the AWS Marketplace to launch and subscribe to official
+Ubuntu Pro images that allow users to pay for additional support.
+These Ubuntu Pro images can also be found in the SSM parameter store.
```
Comment (k-dimple) [inline aws/ec2/ec2-how-to/find-ubuntu-images.rst:106]: Delete the above statement.

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+
+The Marketplace images can be found in the SSM parameter store:
```
Comment (k-dimple) [inline aws/ec2/ec2-how-to/find-ubuntu-images.rst:109]: Change this statement to:
"All the above mentioned marketplace images can also be found in the SSM parameter store:"

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+.. code-block::
+
+   aws ssm get-parameter --name /aws/service/marketplace/$IDENTIFIER/latest
+
+* IDENTIFIER one of the identifiers (starting with `prod-`)
+
```
Comment (k-dimple) [inline aws/ec2/ec2-how-to/find-ubuntu-images.rst:116]: Change to:
IDENTIFIER: use one of the following identifiers (starting with `prod-`)

File: .wordlist.txt [hunk 4]
```diff
@@ -100,3 +100,18 @@ Tanzu
 OpenShift
 RHCOS
 RHEL
+SSM
+OwnerId
+aws
+GovCloud
+minimized
```
Comment (k-dimple) [inline .wordlist.txt:107]: remove this and use 'minimised' instead

File: .wordlist.txt [hunk 5]
```diff
@@ -100,3 +100,18 @@ Tanzu
...
+Anbox
+ImageLocation
+VIRT
+hvm
+pv
+ebs
+gp
+io
+xenial
+discoverable
```
Comment (k-dimple) [inline .wordlist.txt:117]: won't be needed if the change suggested below is used.

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+   
+   ubuntu/$PRODUCT/$RELEASE/stable/$SERIAL/$ARCH/$VIRT_TYPE/$VOL_TYPE/ami-id
+
+
+Find EKS images with SSM
+------------------------
+Amazon’s Elastic Kubernetes Service (EKS) is a managed Kubernetes service provided
+by AWS that let users run Kubernetes applications in the cloud or on-premises.
+
+Canonical provides minimized Ubuntu images customised for use with EKS
```
Comment (k-dimple) [inline aws/ec2/ec2-how-to/find-ubuntu-images.rst:53]: minimized -> minimised

**Review by k-dimple — CHANGES_REQUESTED** [review]

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+   ubuntu/$PRODUCT/$RELEASE/stable/current/$ARCH/$VIRT_TYPE/$VOL_TYPE/ami-id
+
+* PRODUCT: `server`, `server-minimal` or `pro-server`
+* RELEASE: `jammy`, `22.04`, `focal`, `20.04`, `bionic`, `18.04`, `xenial`, or `16.04`
+* ARCH: `amd64` or `arm64`
+* VIRT_TYPE: `pv` or `hvm`
+* VOL_TYPE: `ebs-gp2`, `ebs-io1`, `ebs-standard`, or `instance-store`
+
+The given serial for an image (e.g., `20210222`) is also uploaded in place of `current`:
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:41]: Done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+.. code-block::
+   
+   ubuntu/$PRODUCT/$RELEASE/stable/$SERIAL/$ARCH/$VIRT_TYPE/$VOL_TYPE/ami-id
+
+
+Find EKS images with SSM
+------------------------
+Amazon’s Elastic Kubernetes Service (EKS) is a managed Kubernetes service provided
+by AWS that let users run Kubernetes applications in the cloud or on-premises.
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:51]: done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst (hunk 2 above)
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:109]: Done

File: .wordlist.txt (hunk 4 above)
Comment (toabctl) [reply .wordlist.txt:107]: done

File: .wordlist.txt (hunk 5 above)
Comment (toabctl) [reply .wordlist.txt:117]: Done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst (hunk 6 above)
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:53]: Done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 9]
```diff
@@ -0,0 +1,122 @@
+Find Ubuntu images on AWS EC2
+=============================
+
+On EC2, cloud images are referred to as Amazon Machine Images (AMIs).
+Canonical produces a wide variety of images to support numerous features found on EC2:
+
+* Generally, all images use Elastic Block Storage (EBS) and HVM virtualisation types.
+  Older releases may also support PV and instance-store, but users benefit from the
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst]: Done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+
+.. code-block::
+
+   ubuntu/eks/$RELEASE/$K8S_VERSION/stable/current/$ARCH/$VIRT_TYPE/$VOL_TYPE/ami-id
+
+Ownership verification
+----------------------
+Users can verify that an AMI was published by Canonical by ensuring
+the `OwnerId` field of an image is set to the expected value. Those values are:
+
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:74]: done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+This ID is stored in SSM and is discoverable by running:
+
+.. code-block::
+
+   aws ssm get-parameters --names /aws/service/canonical/meta/publisher-id
+
+
+With the value returned by that command, users can then run the `describe-images`
+command against an AMI ID and verify the `OwnerId` field matches the above ID:
+
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst:87]: done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+and look for `aws-marketplace/ubuntu` in the `ImageLocation` field.
+
+
+AWS Marketplace
+---------------
+AWS Marketplace is a digital catalogue with thousands of software listings
+from independent software vendors that make it easy to find, test, buy, and
+deploy software that runs on AWS.
+Canonical maintains `image listings <https://aws.amazon.com/marketplace/seller-profile?id=565feec9-3d43-413e-9760-c651546613f2>`_ for recent Ubuntu releases and special flavors (e.g. `Anbox`, `Pro`, `EKS`) on AWS Marketplace.
+
```
[the remark this answers is not in the document]
Comment (toabctl) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst]: done

File: aws/ec2/ec2-how-to/find-ubuntu-images.rst [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+=============================
+
+On EC2, cloud images are referred to as Amazon Machine Images (AMIs).
+Canonical produces a wide variety of images to support numerous features found on EC2:
+
+* Generally, all images use Elastic Block Storage (EBS) and HVM virtualisation types.
+  Older releases may also support PV and instance-store, but users benefit from the
+  newer storage and virtualisation technologies.
+* Standard server images, as well as minimal images for amd64, and arm64 images for
+  the standard server set.
```
[the remark this answers is not in the document]
Comment (k-dimple) [reply aws/ec2/ec2-how-to/find-ubuntu-images.rst]: Yes, that sounds fine.

**Review by k-dimple — APPROVED** [review]

**Pull request merged**